homeassistant/components/picnic/todo.py

```diff
@@ -0,0 +1,75 @@
+"""Definition of Picnic shopping cart."""
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from homeassistant.components.todo import TodoItem, TodoItemStatus, TodoListEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import CONF_COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Picnic shopping cart todo platform config entry."""
+    picnic_coordinator = hass.data[DOMAIN][config_entry.entry_id][CONF_COORDINATOR]
+
+    # Add an entity shopping card
+    async_add_entities([PicnicCart(picnic_coordinator, config_entry)])
+
+
+class PicnicCart(TodoListEntity, CoordinatorEntity):
```

Comment:
Please invert the inheritance order.

Reply:
Also add a type annotation to the generic `CoordinatorEntity` what coordinator type it holds.